core/src/main/scala/kafka/utils/Logging.scala

```diff
@@ -17,106 +17,91 @@
+    case Failure(_) => logger.info("Couldn't register kafka:type=kafka.Log4jController MBean")
   }
+}
 
-  def isDebugEnabled: Boolean = logger.isDebugEnabled
+trait Logging extends LazyLogging {
+  def loggerName: String = this.getClass.getName
 
-  def isTraceEnabled: Boolean = logger.isTraceEnabled
+  protected var logIdent: String = _
 
-  def debug(msg: => String): Unit = {
-    if (logger.isDebugEnabled())
-      logger.debug(msgWithLogIdent(msg))
-  }
-  def debug(e: => Throwable): Any = {
-    if (logger.isDebugEnabled())
-      logger.debug(logIdent,e)
-  }
-  def debug(msg: => String, e: => Throwable) = {
-    if (logger.isDebugEnabled())
-      logger.debug(msgWithLogIdent(msg),e)
-  }
-  def swallowDebug(action: => Unit) {
-    CoreUtils.swallow(logger.debug, action)
-  }
+  Log4jControllerRegistration
 
-  def info(msg: => String): Unit = {
-    if (logger.isInfoEnabled())
-      logger.info(msgWithLogIdent(msg))
-  }
-  def info(e: => Throwable): Any = {
-    if (logger.isInfoEnabled())
-      logger.info(logIdent,e)
-  }
-  def info(msg: => String,e: => Throwable) = {
-    if (logger.isInfoEnabled())
-      logger.info(msgWithLogIdent(msg),e)
-  }
-  def swallowInfo(action: => Unit) {
-    CoreUtils.swallow(logger.info, action)
-  }
+  protected def msgWithLogIdent(msg: String): String =
+    if (logIdent == null) msg else logIdent + msg
 
-  def warn(msg: => String): Unit = {
-    logger.warn(msgWithLogIdent(msg))
-  }
-  def warn(e: => Throwable): Any = {
-    logger.warn(logIdent,e)
-  }
-  def warn(msg: => String, e: => Throwable) = {
-    logger.warn(msgWithLogIdent(msg),e)
-  }
-  def swallowWarn(action: => Unit) {
-    CoreUtils.swallow(logger.warn, action)
-  }
-  def swallow(action: => Unit) = swallowWarn(action)
+  def trace(msg: => String): Unit = logger.trace(msgWithLogIdent(msg))
 
-  def error(msg: => String): Unit = {
-    logger.error(msgWithLogIdent(msg))
-  }		
-  def error(e: => Throwable): Any = {
-    logger.error(logIdent,e)
-  }
-  def error(msg: => String, e: => Throwable) = {
-    logger.error(msgWithLogIdent(msg),e)
-  }
-  def swallowError(action: => Unit) {
-    CoreUtils.swallow(logger.error, action)
-  }
+  def trace(e: => Throwable): Any = logger.trace(logIdent,e)
 
-  def fatal(msg: => String): Unit = {
-    logger.fatal(msgWithLogIdent(msg))
-  }
-  def fatal(e: => Throwable): Any = {
-    logger.fatal(logIdent,e)
-  }	
-  def fatal(msg: => String, e: => Throwable) = {
-    logger.fatal(msgWithLogIdent(msg),e)
-  }
+  def trace(msg: => String, e: => Throwable): Unit = logger.trace(msgWithLogIdent(msg),e)
+
+  def swallowTrace(action: => Unit): Unit = CoreUtils.swallow(logger.underlying.trace, action)
+
+  def isDebugEnabled: Boolean = logger.underlying.isDebugEnabled
+
+  def isTraceEnabled: Boolean = logger.underlying.isTraceEnabled
+
+  def debug(msg: => String): Unit = logger.debug(msgWithLogIdent(msg))
+
+  def debug(e: => Throwable): Any = logger.debug(logIdent,e)
```

Comment:
I'll make a developer note as the return type should be Unit but since `e: => Throwable` will be compiled to a Function which will be affected by type erasure thus removing the `String` or `Throwable` types making the functions the same. The difference in the return type will make a difference as it is part of the function's signature.

Reply:
We don't seem to use this and most of the other methods that take `=> Throwable`, so I'd just remove them. There seems to be a single usage of the `fatal` version, but we can probably just use the variant that also takes a message.

Reply:
Ok, I'll get rid of them.